openlabcmd/openlabcmd/plugins/nodepool/network.py

```diff
@@ -39,15 +39,29 @@ def check(self):
             self.reasons.append(Recover.NETWORK_SUBNET)
             return
 
-        net = 'openstack --os-cloud %s router show openlab-router' % self.cloud
-        ret, res = subprocess.getstatusoutput(net)
+        external_net = 'openstack --os-cloud %s network list --external ' \
+                       '-f value -c ID' % self.cloud
+        ret, res = subprocess.getstatusoutput(external_net)
+        if ret != 0:
+            self.failed = True
+            self.reasons.append('Failed to get a external network.')
+            return
+
+        ext_net_id = res
+
+        router = 'openstack --os-cloud %s router show ' \
+                 'openlab-router' % self.cloud
+        ret, res = subprocess.getstatusoutput(router)
         if ret != 0:
             self.failed = True
-            if "More than one Openlab router exists" in res:
+            if "More than one Router exists" in res:
                 self.reasons.append(res)
             else:
                 self.reasons.append(Recover.ROUTER)
                 self.reasons.append(Recover.ROUTER_SUBNET_INTERFACE)
+                self.reasons.append(Recover.ROUTER_EXTERNAL_GW)
```

Comment:
L25 and L40 will return directly, and the router check will be skiped, that means in an empty env, we should do openlab check twice, does this meet your mind?

Reply:
I know, but network and router resources have no relationship. So I need to bring nested check into every step?? When we do recover process, It checks the network is not exist, so we set Recover.Netowrk into Recovering process, but we don't know if Router exists, right? we should cover these cases.